### PR TITLE
KAFKA-16617: Include kraft handler in advertised listener doc

### PR DESCRIPTION
**Pull request by Owen-CH-Leung**
As per [KAFKA-16617](https://issues.apache.org/jira/browse/KAFKA-16617), this PR revised the documentation for `advertised.listeners` to include kraft handler. 

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java [hunk 1]
```diff
@@ -58,13 +58,17 @@ public class SocketServerConfigs {
...
     public static final String ADVERTISED_LISTENERS_DOC = String.format(
-            "Listeners to publish to ZooKeeper for clients to use, if different than the <code>%s</code> config property." +
-                    " In IaaS environments, this may need to be different from the interface to which the broker binds." +
-                    " If this is not set, the value for <code>%1$1s</code> will be used." +
-                    " Unlike <code>%1$1s</code>, it is not valid to advertise the 0.0.0.0 meta-address.%n" +
-                    " Also unlike <code>%1$1s</code>, there can be duplicated ports in this property," +
-                    " so that one listener can be configured to advertise another listener's address." +
-                    " This can be useful in some cases where external load balancers are used.", LISTENERS_CONFIG);
+            "Specifies the listener addresses that the Kafka brokers will advertise to clients and other brokers." +
+            " The config is useful where the actual listener configuration <code>%s</code> does not represent the addresses that clients should" +
```
Comment (showuon) [inline server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java:62]: Should we use `<code>%1$1s</code>` as below?

File: server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java [hunk 2]
```diff
@@ -58,13 +58,17 @@ public class SocketServerConfigs {
...
+            " use to connect, such as in cloud environments. In environments using ZooKeeper, these addresses are published to ZooKeeper." +
+            " In KRaft mode, these addresses are managed internally by the Kafka brokers themselves. Regardless of the operating mode, this config" +
```
Comment (showuon) [inline server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java:64]: > In KRaft mode, these addresses are managed internally by the Kafka brokers themselves.

Is this correct? I mean, yes, they have metadataCache, like in ZK, the broker also has metadataCache. But in ZK, we'll publish to ZK, so in KRaft, where will we publish the advertised listener to?

**Review by showuon — COMMENTED** [review]
@Owen-CH-Leung , thanks for the PR. Left some comments.

File: server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java (hunk 2 above)
Comment (Owen-CH-Leung) [reply server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java:64]: you mean - while such information will be stored at metadataCache, they will also be published to controller ?

**Comment by chia7712** [comment]
@Owen-CH-Leung any updates?

**Comment by Owen-CH-Leung** [comment]
@chia7712 I'll close this PR and let another contributor work on it

**Pull request closed without merging**